controllers/helmchart_controller.go

```diff
@@ -451,6 +456,15 @@ func (r *HelmChartReconciler) buildFromHelmRepository(ctx context.Context, obj *
 		}
 	}
 
+	// Try to retrieve the repository index from the cache
+	if r.Cache != nil {
+		if index, found := r.Cache.Get(r.Storage.LocalPath(*repo.GetArtifact())); err == nil {
+			if found {
```

Comment:
would be nice to have a prometheus metric here to help administrators adjust cache size based on hit ratio.